api/cpp/esp-idf/slint/README.md

```diff
@@ -11,81 +11,110 @@ It has been tested on ESP32-S3 devices.
+     /* Set display brightness to 100% */
+    bsp_display_backlight_on();
+
+    std::optional<esp_lcd_touch_handle_t> touch_handle;
+
+    /* Allocate a drawing buffer */
+    static std::vector<slint::platform::Rgb565Pixel> buffer(BSP_LCD_H_RES * BSP_LCD_V_RES);
+
+    /* Initialize Slint's ESP platform support*/
+    slint_esp_init(slint::PhysicalSize({ BSP_LCD_H_RES, BSP_LCD_V_RES }), panel_handle,
+                                       touch_handle, buffer);
+    /* Instantiate the UI */
+    auto ui = AppWindow::create();
+    /* Show it on the screen and run the event loop */
+    ui->run();
+}
 ```
-error: the `-Z` flag is only accepted on the nightly channel of Cargo, but this is the `stable` channel
+1. Create `main/appwindow.slint` with the following contents:
 ```
-
-this means you are not using the right toolchain.
-
-
-## Usage
-
-By using this component, the `Slint::Slint` CMake target is linked to your application and you can access the entire functionality of the
-[Slint C++ API](https://slint.dev/docs/cpp).
-
-In addition, this component provides the `slint-esp.h` header file, which provides a a function to initialize the Slint backend
-that must be called before setting up the UI.
-Behind the scene, it will use `esp_lcd_panel_draw_bitmap` to render to the screen,
-and is base on [ESP LCD Touch](https://components.espressif.com/components/espressif/esp_lcd_touch) for the touch events.
-
-Use this platform implementation by instantiating it with a `esp_lcd_panel_handle_t`, an optional `esp_lcd_touch_handle_t`, and a pointer to one
-or two `slint::platform::Rgb565Pixel` frame buffers. Next, register the instance with the Slint run-time library by calling `slint_esp_init()`:
-
-```cpp
-#include "slint-esp.h"
-// ...
-
-
-// Initialize the panel and touch handle
-esp_lcd_panel_handle_t panel_handle = NULL;
-bsp_display_new(&bsp_disp_cfg, &panel_handle, &io_handle);
-bsp_display_backlight_on();
-esp_lcd_touch_handle_t touch_handle = NULL;
-bsp_touch_new(&bsp_touch_cfg, &touch_handle);
-
-// Allocate a frame buffer
-static std::vector<slint::platform::Rgb565Pixel> buffer(BSP_LCD_H_RES * BSP_LCD_V_RES);
-
-// initialize the Slint ESP backend
-slint_esp_init(slint::PhysicalSize({ BSP_LCD_H_RES, BSP_LCD_V_RES }), panel_handle,
-                                   touch_handle, buffer);
-
+import { VerticalBox, AboutSlint } from "std-widgets.slint";
+export component AppWindow inherits Window {
+    VerticalBox {
+        AboutSlint {}
+        Text {
+            text: "Hello World";
+            font-size: 18px;
+            horizontal-alignment: center;
+        }
+    }
+}
 ```
-
-Alternatively, you can implement your own sub-class of `slint::platform::Platform` to drive the screen and handle input events.
-
-Next, integrate your `.slint` files into the build by compiling them to C++ and linking them to your component:
-
+1. Edit `main/CMakeLists.txt` to adjust for the new `slint-hello-world.cpp`, add `slint` as required component,
```

Comment:
this goes back to 1.  (also when previewing in github

Reply:
Oops, thanks for noticing